### PR TITLE
Replace issue templates with issue forms

### PR DESCRIPTION
**Pull request by steveward**
## Description

This PR replaces our existing issue templates with issue forms, which should hopefully cut down on the number of empty issues we receive. 

Thanks to @KendallDoesCoding in https://github.com/desktop/desktop/pull/13517 for helping initiate this process. I've stayed fairly true to our existing issue templates -- please feel free to leave feedback if anything can be improved. 

## Release notes

n/a

Notes:

### CODE REVIEW

**Comment by steveward** [comment]
Links for how they render for anyone reviewing:

https://github.com/desktop/desktop/blob/add-issue-forms/.github/ISSUE_TEMPLATE/bug_report.yaml
https://github.com/desktop/desktop/blob/add-issue-forms/.github/ISSUE_TEMPLATE/feature_request.yaml

File: .github/ISSUE_TEMPLATE/feature_request.yaml [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+  - type: markdown
+    attributes:
+      value: |
+        Thanks for submitting a feature request! This issue tracker is for [GitHub Desktop](https://desktop.github.com).  If you have a feature request for github.com please visit the [GitHub public feedback discussions repository](https://github.com/github/feedback/discussions).
+  - type: checkboxes
+    id: existing-feature-requests
+    attributes:
+      label: Search for existing feature requests
+      options:
+        - label: I have searched the issue tracker and confirmed there are no existing issues for the feature I am requesting.
```
Comment (billygriffin) [inline .github/ISSUE_TEMPLATE/feature_request.yaml]:
```suggestion
        - label: I have searched the issue tracker and confirmed that I was unable to find existing issues for the feature I am requesting.
```

File: .github/ISSUE_TEMPLATE/bug_report.yaml [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+  - type: markdown
+    attributes:
+      value: |
+        Thanks for filing a bug report! This issue tracker is for [GitHub Desktop](https://desktop.github.com). If you are experiencing issues with the Linux fork of GitHub Desktop please open an issue [in that repository](https://github.com/shiftkey/desktop). If you are experiencing issues with github.com please [contact GitHub Support](https://support.github.com/).
+  - type: checkboxes
+    id: existing-issues
+    attributes:
+      label: Search existing issues
+      options:
+        - label: I have searched the issue tracker and confirmed there are no existing issues for the problem I am experiencing.
```
Comment (billygriffin) [inline .github/ISSUE_TEMPLATE/bug_report.yaml]:
```suggestion
        - label: I have searched the issue tracker and confirmed that I was unable to find existing issues for the problem I am experiencing.
```

File: .github/ISSUE_TEMPLATE/bug_report.yaml [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+    attributes:
+      label: Release version
+      description: Open the 'About GitHub Desktop' menu item to see the GitHub Desktop version.
+    validations:
+      required: true
+  - type: input
+    id: operating-system
+    attributes:
+      label: Operating system
+      description: Enter the specific operating system version you are running.
```
Comment (billygriffin) [inline .github/ISSUE_TEMPLATE/bug_report.yaml]:
```suggestion
      description: Enter the specific operating system version you are running (example: Windows 10)
```

**Review by billygriffin — DISMISSED** [review]
This looks great overall @steveward, thank you! I requested a couple minor changes just to make things a bit clearer and slightly more friendly for people, but overall I think this is going to be a nice improvement.

**Review by KendallDoesCoding — COMMENTED** [review]
.yaml is my suggestion but yml works well too.

**Review by tidy-dev — DISMISSED** [review]
✨  TIL you could do this with a yaml for forms and such. This is awesome!

**Comment by KendallDoesCoding** [comment]
> ✨ TIL you could do this with a yaml for forms and such. This is awesome!

Yup, indeed! Hence, my suggestion.

File: .github/ISSUE_TEMPLATE/bug_report.yaml [hunk 4]
```diff
@@ -0,0 +1,62 @@
+name: 🐛 Bug Report
+description: File a bug report
+body:
+  - type: markdown
+    attributes:
+      value: |
+        Thanks for filing a bug report! This issue tracker is for [GitHub Desktop](https://desktop.github.com). If you are experiencing issues with the Linux fork of GitHub Desktop please open an issue [in that repository](https://github.com/shiftkey/desktop). If you are experiencing issues with github.com please [contact GitHub Support](https://support.github.com/).
```
Comment (KendallDoesCoding) [inline .github/ISSUE_TEMPLATE/bug_report.yaml]:
```suggestion
        Thanks for filing a bug report! This issue tracker is for [GitHub Desktop](https://desktop.github.com). If you are experiencing issues with the Linux fork of GitHub Desktop please open an issue [in this repository](https://github.com/shiftkey/desktop). If you are experiencing issues with github.com please [contact GitHub Support](https://support.github.com/).
```

Update grammer!

**Review by KendallDoesCoding — COMMENTED** [review]
Minor grammer edit.

File: .github/ISSUE_TEMPLATE/bug_report.yaml (hunk 4 above)
Comment (Rexogamer) [reply .github/ISSUE_TEMPLATE/bug_report.yaml]: I think this suggestion makes this slightly less clear - maybe something like this?
```suggestion
        Thanks for filing a bug report! This issue tracker is for [GitHub Desktop](https://desktop.github.com). If you are experiencing issues with the Linux fork of GitHub Desktop please open an issue [in its repository](https://github.com/shiftkey/desktop). If you are experiencing issues with github.com please [contact GitHub Support](https://support.github.com/).
```

Comment (KendallDoesCoding) [reply .github/ISSUE_TEMPLATE/bug_report.yaml]: Yup, either way works. Thanks for the alternative suggestion @Rexogamer

**Review by KendallDoesCoding — COMMENTED** [review]
"YAML syntax error: (): mapping values are not allowed in this context at line 33 column 88. Learn more."

**Review by steveward — COMMENTED** [review]
I think this is ready to go!

**Review by tidy-dev — APPROVED** [review]

**Comment by KendallDoesCoding** [comment]
Looks great, thanks for following my suggestion @desktop team.

**Review by ramsemune123 — COMMENTED** [review]
[x3CaaKRWyzU.csv](https://github.com/desktop/desktop/files/8298625/x3CaaKRWyzU.csv)

**Pull request merged**